tests/testthat/test-depth.R

```diff
@@ -0,0 +1,19 @@
+context("calc_depth requires and returns reasonable values")
+
+test_that("calc_depth checks & returns values & units as expected", {
+  # basic numbers, with & without defaults
+  expect_is(calc_depth(Q=7.3), "numeric")
+  expect_equal(calc_depth(Q=7.3, c=1, f=0.5), 7.3^0.5)
+  expect_equal(calc_depth(Q=rep(100, 3), c=seq(0.4, 0.6, 0.1), f=c(0.25, 0.29, 0.33)), c(1.26, 1.9, 2.74), tol=0.1)
+  
+  # with units - throws warnings rather than errors in several cases
+  library(unitted)
+  expect_error(expect_warning(calc_depth(Q=1:10, f=u(0.36)), "not unitted"), "Unexpected units")
+  expect_warning(calc_depth(Q=u(1:10, "m^3 s^-1"), c=0.36), "unknown depth units")
```

Comment:
Worth discussing. The function works correctly, i.e., the units of the outputs match those of the inputs, no matter what the units of c are. So you could legitimately have a constant for c in meters or in centimeters. But some units of c would be strange - empty units, "gremlins", etc. So I thought I'd just throw a warning but let the user proceed anyway if they want. I dunno - maybe that's too lax relative to the rest of what we're doing?